shared/src/main/scala/mouse/any.scala

```diff
@@ -0,0 +1,12 @@
+package mouse
+
+trait AnySyntax {
+  implicit final def optionSyntaxMouse[A](oa: A): AnyOps[A] = new AnyOps(oa)
+}
+
+final class AnyOps[A](val oa: A) extends AnyVal {
+  def |>[B] (f: A => B) = f(oa)
+  def thrush[B] (f: A => B) = f(oa)
+
+  def ignore = mouse ignore oa
```

Comment:
@stephen-lazaro if `ignore` is now imported via package object, can we drop this enrichment? 

There's a real cost if pimping extra methods onto `Any` and `ignore` is quite a common name. Users of Mouse may already use an `ignore` operation defined elsewhere, in which case they'll get an ambiguous implicits conflict.

A classic example was in the Scala early years, when they decided to pimp `+` onto everything (to help with `toString`). It ended up obstructing people trying to build libraries using numeric typeclasses.